File: backend/src/Entity/EmailVerificationCode.php
<?php

namespace App\Entity;

use App\Repository\EmailVerificationCodeRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: EmailVerificationCodeRepository::class)]
class EmailVerificationCode
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'code')]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private ?User $user = null;

    #[ORM\Column(length: 255)]
    private ?string $code = null;

    #[ORM\Column]
    private ?\DateTimeImmutable $expiresAt = null;

    #[ORM\Column(nullable: true)]
    private ?\DateTimeImmutable $usedAt = null;

    #[ORM\Column(type: 'integer')]
    private int $resendAttempts = 0;

    #[ORM\Column(type: 'integer')]
    private int $verifyAttempts = 0;

    #[ORM\Column(type: 'datetime_immutable')]
    private ?\DateTimeImmutable $lastSentAt = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): static
    {
        $this->user = $user;

        return $this;
    }

    public function getCode(): ?string
    {
        return $this->code;
    }

    public function setCode(string $code): static
    {
        $this->code = $code;

        return $this;
    }

    public function getExpiresAt(): ?\DateTimeImmutable
    {
        return $this->expiresAt;
    }

    public function setExpiresAt(\DateTimeImmutable $expiresAt): static
    {
        $this->expiresAt = $expiresAt;

        return $this;
    }

    public function getUsedAt(): ?\DateTimeImmutable
    {
        return $this->usedAt;
    }

    public function setUsedAt(?\DateTimeImmutable $usedAt): static
    {
        $this->usedAt = $usedAt;

        return $this;
    }
    public function getLastSentAt(): ?\DateTimeImmutable
    {
        return $this->lastSentAt;
    }

    public function setLastSentAt(?\DateTimeImmutable $lastSentAt): static
    {
        $this->lastSentAt = $lastSentAt;

        return $this;
    }

    public function getResendAttempts(): int
    {
        return $this->resendAttempts;
    }

    public function setResendAttempts(int $resendAttempts): static
    {
        $this->resendAttempts = $resendAttempts;

        return $this;
    }

    public function getVerifyAttempts(): int
    {
        return $this->verifyAttempts;
    }

    public function setVerifyAttempts(int $verifyAttempts): static
    {
        $this->verifyAttempts = $verifyAttempts;

        return $this;
    }
}

File: frontend/src/main/java/com/innertrack/model/EmailVerificationCode.java
package com.innertrack.model;

import java.time.LocalDateTime;

public class EmailVerificationCode {
    private int id;
    private User user;
    private String code;
    private LocalDateTime expiresAt;
    private LocalDateTime usedAt;
    private int resendAttempts;
    private int verifyAttempts;
    private LocalDateTime lastSentAt;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public LocalDateTime getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(LocalDateTime expiresAt) {
        this.expiresAt = expiresAt;
    }

    public LocalDateTime getUsedAt() {
        return usedAt;
    }

    public void setUsedAt(LocalDateTime usedAt) {
        this.usedAt = usedAt;
    }

    public int getResendAttempts() {
        return resendAttempts;
    }

    public void setResendAttempts(int resendAttempts) {
        this.resendAttempts = resendAttempts;
    }

    public int getVerifyAttempts() {
        return verifyAttempts;
    }

    public void setVerifyAttempts(int verifyAttempts) {
        this.verifyAttempts = verifyAttempts;
    }

    public LocalDateTime getLastSentAt() {
        return lastSentAt;
    }

    public void setLastSentAt(LocalDateTime lastSentAt) {
        this.lastSentAt = lastSentAt;
    }
}
